### ISSUE TITLE
utah-0.1.2 test failure running linker on 1.16 beta

### ISSUE THREAD

**Issue by brson** [rust-lang/rust#39830]
```
brian@ip-10-145-43-250:/mnt2/dev/utah⟫ git remote -v
origin  https://github.com/pegasos1/utah (fetch)
origin  https://github.com/pegasos1/utah (push)
brian@ip-10-145-43-250:/mnt2/dev/utah⟫ git log -1
commit 99c244a1e811128a8279616ea8d6024ed3c17823
Author: Suchin <sgrangan01@gmail.com>
Date:   Wed Dec 28 11:27:03 2016 -0800

    Update README.md
```

```
brian@ip-10-145-43-250:/mnt2/dev⟫ rustc +beta -Vv
rustc 1.16.0-beta.1 (5276ba72e 2017-01-31)
binary: rustc
commit-hash: 5276ba72e8fe13597200254447fce9cb3783c583
commit-date: 2017-01-31
host: x86_64-unknown-linux-gnu
release: 1.16.0-beta.1
LLVM version: 3.9
```

```
   Doc-tests utah

running 20 tests
test _0 ... ignored
test _1 ... ignored
test _10 ... ignored
test _11 ... ignored
test _2 ... ignored
test _3 ... ignored
test _4 ... ignored
test _5 ... ignored
test _6 ... ignored
test _7 ... ignored
test _8 ... ignored
test _9 ... ignored
test implement::impl_constructor_generic::DataFrame<T>::df_iter_mut_0 ... ok
test implement::impl_constructor_generic::DataFrame<T>::from_array_0 ... ok
test implement::impl_constructor_generic::DataFrame<T>::columns_0 ... FAILED
test implement::impl_constructor_generic::DataFrame<T>::df_iter_0 ... FAILED
test implement::impl_constructor_generic::DataFrame<T>::new_0 ... ok
test implement::impl_constructor_generic::DataFrame<T>::index_0 ... FAILED
test implement::impl_constructor_generic::DataFrame<T>::index_1 ... FAILED
test implement::impl_constructor_generic::DataFrame<T>::new_1 ... ok

failures:

---- implement::impl_constructor_generic::DataFrame<T>::columns_0 stdout ----
        error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-m64" "-L" "/home/brian/.rustup/toolchains/beta-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "/tmp/rustdoctest.PrTrjBAF18uG/rust_out.0.o" "-o" "/tmp/rustdoctest.PrTrjBAF18uG/rust_out" "-Wl,--gc-sections" "-pie" "-nodefaultlibs" "-L" "/mnt2/dev/utah/target/debug/deps" "-L" "/home/brian/.rustup/toolchains/beta-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "-Wl,-Bdynamic" "/mnt2/dev/utah/target/debug/deps/libutah-7ebd1627d8f4b31c.rlib" "/mnt2/dev/utah/target/debug/deps/libndarray_rand-bd14357e2975e921.rlib" "/mnt2/dev/utah/target/debug/deps/libnum-4f5490b7b4313ad0.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_iter-a1bfa3b9618fdfe0.rlib" "/mnt2/dev/utah/target/debug/deps/liberror_chain-51998ce5824b8fe7.rlib" "/mnt2/dev/utah/target/debug/deps/libbacktrace-4bd69dcf892822ef.rlib" "/mnt2/dev/utah/target/debug/deps/libbacktrace_sys-58044020ec178381.rlib" "/mnt2/dev/utah/target/debug/deps/libdbghelp-80abcaa2ef08d9db.rlib" "/mnt2/dev/utah/target/debug/deps/libwinapi-a5898d7aceb63fac.rlib" "/mnt2/dev/utah/target/debug/deps/librustc_demangle-ffe0cd46130fc1c4.rlib" "/mnt2/dev/utah/target/debug/deps/libndarray-a08975650c3bc21a.rlib" "/mnt2/dev/utah/target/debug/deps/libmatrixmultiply-f1139eb0050ad26b.rlib" "/mnt2/dev/utah/target/debug/deps/librawpointer-dc9eeb50dff13477.rlib" "/mnt2/dev/utah/target/debug/deps/libitertools-ce89a6802ef9c3ff.rlib" "/mnt2/dev/utah/target/debug/deps/libeither-f2ccbfaf1069b354.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_complex-2d6fd57f39d31306.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_rational-e1a0d96e4c2f73b6.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_bigint-a0b0a32a0399692f.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_integer-f617fb54b59f25d7.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_traits-51f88ce8f32c0283.rlib" "/mnt2/dev/utah/target/debug/deps/librand-c9d9fbdab2355ee4.rlib" "/mnt2/dev/utah/target/debug/deps/liblibc-13f441799a82421b.rlib" "/mnt2/dev/utah/target/debug/deps/libkernel32-835ed4d4f4dc2d3e.rlib" "/mnt2/dev/utah/target/debug/deps/libcfg_if-5cf5b4dd25466da1.rlib" "/mnt2/dev/utah/target/debug/deps/libcsv-409fb1ac4ede067a.rlib" "/mnt2/dev/utah/target/debug/deps/librustc_serialize-2c0b16b7ceb9fdcb.rlib" "/mnt2/dev/utah/target/debug/deps/libbyteorder-fab4d718c73e8b35.rlib" "-L" "/home/brian/.rustup/toolchains/beta-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-l" "std-dfc89321dbee872e" "/home/brian/.rustup/toolchains/beta-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-a27e0b02588f3694.rlib" "-l" "util" "-l" "dl" "-l" "rt" "-l" "pthread" "-l" "gcc_s" "-l" "pthread" "-l" "c" "-l" "m" "-l" "rt" "-l" "util"
  = note: /mnt2/dev/utah/target/debug/deps/libbacktrace-4bd69dcf892822ef.rlib(backtrace-4bd69dcf892822ef.0.o): In function `backtrace::symbolize::libbacktrace::init_state::{{closure}}':
          /home/brian/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.0/src/symbolize/mod.rs:159: undefined reference to `backtrace_create_state'
          /mnt2/dev/utah/target/debug/deps/libbacktrace-4bd69dcf892822ef.rlib(backtrace-4bd69dcf892822ef.0.o): In function `backtrace::symbolize::libbacktrace::resolve':
          /home/brian/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.0/src/symbolize/mod.rs:159: undefined reference to `backtrace_pcinfo'
          /home/brian/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.0/src/symbolize/mod.rs:159: undefined reference to `backtrace_syminfo'
          collect2: error: ld returned 1 exit status


error: aborting due to previous error

thread 'rustc' panicked at 'Box<Any>', /buildslave/rust-buildbot/slave/beta-dist-rustc-linux/build/src/librustc_errors/lib.rs:453
note: Run with `RUST_BACKTRACE=1` for a backtrace.
thread 'rustc' panicked at 'couldn't compile the test', /buildslave/rust-buildbot/slave/beta-dist-rustc-linux/build/src/librustdoc/test.rs:265

---- implement::impl_constructor_generic::DataFrame<T>::df_iter_0 stdout ----
        warning: unused variable: `df_iter`, #[warn(unused_variables)] on by default
 --> <anon>:6:5
  |
6 | let df_iter = df.df_iter(UtahAxis::Row);
  |     ^^^^^^^

error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-m64" "-L" "/home/brian/.rustup/toolchains/beta-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "/tmp/rustdoctest.knVrbrD9nkWB/rust_out.0.o" "-o" "/tmp/rustdoctest.knVrbrD9nkWB/rust_out" "-Wl,--gc-sections" "-pie" "-nodefaultlibs" "-L" "/mnt2/dev/utah/target/debug/deps" "-L" "/home/brian/.rustup/toolchains/beta-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "-Wl,-Bdynamic" "/mnt2/dev/utah/target/debug/deps/libutah-7ebd1627d8f4b31c.rlib" "/mnt2/dev/utah/target/debug/deps/libndarray_rand-bd14357e2975e921.rlib" "/mnt2/dev/utah/target/debug/deps/libnum-4f5490b7b4313ad0.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_iter-a1bfa3b9618fdfe0.rlib" "/mnt2/dev/utah/target/debug/deps/liberror_chain-51998ce5824b8fe7.rlib" "/mnt2/dev/utah/target/debug/deps/libbacktrace-4bd69dcf892822ef.rlib" "/mnt2/dev/utah/target/debug/deps/libbacktrace_sys-58044020ec178381.rlib" "/mnt2/dev/utah/target/debug/deps/libdbghelp-80abcaa2ef08d9db.rlib" "/mnt2/dev/utah/target/debug/deps/libwinapi-a5898d7aceb63fac.rlib" "/mnt2/dev/utah/target/debug/deps/librustc_demangle-ffe0cd46130fc1c4.rlib" "/mnt2/dev/utah/target/debug/deps/libndarray-a08975650c3bc21a.rlib" "/mnt2/dev/utah/target/debug/deps/libmatrixmultiply-f1139eb0050ad26b.rlib" "/mnt2/dev/utah/target/debug/deps/librawpointer-dc9eeb50dff13477.rlib" "/mnt2/dev/utah/target/debug/deps/libitertools-ce89a6802ef9c3ff.rlib" "/mnt2/dev/utah/target/debug/deps/libeither-f2ccbfaf1069b354.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_complex-2d6fd57f39d31306.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_rational-e1a0d96e4c2f73b6.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_bigint-a0b0a32a0399692f.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_integer-f617fb54b59f25d7.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_traits-51f88ce8f32c0283.rlib" "/mnt2/dev/utah/target/debug/deps/librand-c9d9fbdab2355ee4.rlib" "/mnt2/dev/utah/target/debug/deps/liblibc-13f441799a82421b.rlib" "/mnt2/dev/utah/target/debug/deps/libkernel32-835ed4d4f4dc2d3e.rlib" "/mnt2/dev/utah/target/debug/deps/libcfg_if-5cf5b4dd25466da1.rlib" "/mnt2/dev/utah/target/debug/deps/libcsv-409fb1ac4ede067a.rlib" "/mnt2/dev/utah/target/debug/deps/librustc_serialize-2c0b16b7ceb9fdcb.rlib" "/mnt2/dev/utah/target/debug/deps/libbyteorder-fab4d718c73e8b35.rlib" "-L" "/home/brian/.rustup/toolchains/beta-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-l" "std-dfc89321dbee872e" "/home/brian/.rustup/toolchains/beta-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-a27e0b02588f3694.rlib" "-l" "util" "-l" "dl" "-l" "rt" "-l" "pthread" "-l" "gcc_s" "-l" "pthread" "-l" "c" "-l" "m" "-l" "rt" "-l" "util"
  = note: /mnt2/dev/utah/target/debug/deps/libbacktrace-4bd69dcf892822ef.rlib(backtrace-4bd69dcf892822ef.0.o): In function `backtrace::symbolize::libbacktrace::init_state::{{closure}}':
          /home/brian/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.0/src/symbolize/mod.rs:159: undefined reference to `backtrace_create_state'
          /mnt2/dev/utah/target/debug/deps/libbacktrace-4bd69dcf892822ef.rlib(backtrace-4bd69dcf892822ef.0.o): In function `backtrace::symbolize::libbacktrace::resolve':
          /home/brian/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.0/src/symbolize/mod.rs:159: undefined reference to `backtrace_pcinfo'
          /home/brian/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.0/src/symbolize/mod.rs:159: undefined reference to `backtrace_syminfo'
          collect2: error: ld returned 1 exit status


error: aborting due to previous error

thread 'rustc' panicked at 'Box<Any>', /buildslave/rust-buildbot/slave/beta-dist-rustc-linux/build/src/librustc_errors/lib.rs:453
thread 'rustc' panicked at 'couldn't compile the test', /buildslave/rust-buildbot/slave/beta-dist-rustc-linux/build/src/librustdoc/test.rs:265

---- implement::impl_constructor_generic::DataFrame<T>::index_0 stdout ----
        error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-m64" "-L" "/home/brian/.rustup/toolchains/beta-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "/tmp/rustdoctest.fleS6jIAZI9w/rust_out.0.o" "-o" "/tmp/rustdoctest.fleS6jIAZI9w/rust_out" "-Wl,--gc-sections" "-pie" "-nodefaultlibs" "-L" "/mnt2/dev/utah/target/debug/deps" "-L" "/home/brian/.rustup/toolchains/beta-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "-Wl,-Bdynamic" "/mnt2/dev/utah/target/debug/deps/libutah-7ebd1627d8f4b31c.rlib" "/mnt2/dev/utah/target/debug/deps/libndarray_rand-bd14357e2975e921.rlib" "/mnt2/dev/utah/target/debug/deps/libnum-4f5490b7b4313ad0.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_iter-a1bfa3b9618fdfe0.rlib" "/mnt2/dev/utah/target/debug/deps/liberror_chain-51998ce5824b8fe7.rlib" "/mnt2/dev/utah/target/debug/deps/libbacktrace-4bd69dcf892822ef.rlib" "/mnt2/dev/utah/target/debug/deps/libbacktrace_sys-58044020ec178381.rlib" "/mnt2/dev/utah/target/debug/deps/libdbghelp-80abcaa2ef08d9db.rlib" "/mnt2/dev/utah/target/debug/deps/libwinapi-a5898d7aceb63fac.rlib" "/mnt2/dev/utah/target/debug/deps/librustc_demangle-ffe0cd46130fc1c4.rlib" "/mnt2/dev/utah/target/debug/deps/libndarray-a08975650c3bc21a.rlib" "/mnt2/dev/utah/target/debug/deps/libmatrixmultiply-f1139eb0050ad26b.rlib" "/mnt2/dev/utah/target/debug/deps/librawpointer-dc9eeb50dff13477.rlib" "/mnt2/dev/utah/target/debug/deps/libitertools-ce89a6802ef9c3ff.rlib" "/mnt2/dev/utah/target/debug/deps/libeither-f2ccbfaf1069b354.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_complex-2d6fd57f39d31306.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_rational-e1a0d96e4c2f73b6.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_bigint-a0b0a32a0399692f.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_integer-f617fb54b59f25d7.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_traits-51f88ce8f32c0283.rlib" "/mnt2/dev/utah/target/debug/deps/librand-c9d9fbdab2355ee4.rlib" "/mnt2/dev/utah/target/debug/deps/liblibc-13f441799a82421b.rlib" "/mnt2/dev/utah/target/debug/deps/libkernel32-835ed4d4f4dc2d3e.rlib" "/mnt2/dev/utah/target/debug/deps/libcfg_if-5cf5b4dd25466da1.rlib" "/mnt2/dev/utah/target/debug/deps/libcsv-409fb1ac4ede067a.rlib" "/mnt2/dev/utah/target/debug/deps/librustc_serialize-2c0b16b7ceb9fdcb.rlib" "/mnt2/dev/utah/target/debug/deps/libbyteorder-fab4d718c73e8b35.rlib" "-L" "/home/brian/.rustup/toolchains/beta-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-l" "std-dfc89321dbee872e" "/home/brian/.rustup/toolchains/beta-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-a27e0b02588f3694.rlib" "-l" "util" "-l" "dl" "-l" "rt" "-l" "pthread" "-l" "gcc_s" "-l" "pthread" "-l" "c" "-l" "m" "-l" "rt" "-l" "util"
  = note: /mnt2/dev/utah/target/debug/deps/libbacktrace-4bd69dcf892822ef.rlib(backtrace-4bd69dcf892822ef.0.o): In function `backtrace::symbolize::libbacktrace::init_state::{{closure}}':
          /home/brian/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.0/src/symbolize/mod.rs:159: undefined reference to `backtrace_create_state'
          /mnt2/dev/utah/target/debug/deps/libbacktrace-4bd69dcf892822ef.rlib(backtrace-4bd69dcf892822ef.0.o): In function `backtrace::symbolize::libbacktrace::resolve':
          /home/brian/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.0/src/symbolize/mod.rs:159: undefined reference to `backtrace_pcinfo'
          /home/brian/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.0/src/symbolize/mod.rs:159: undefined reference to `backtrace_syminfo'
          collect2: error: ld returned 1 exit status


error: aborting due to previous error

thread 'rustc' panicked at 'Box<Any>', /buildslave/rust-buildbot/slave/beta-dist-rustc-linux/build/src/librustc_errors/lib.rs:453
thread 'rustc' panicked at 'couldn't compile the test', /buildslave/rust-buildbot/slave/beta-dist-rustc-linux/build/src/librustdoc/test.rs:265

---- implement::impl_constructor_generic::DataFrame<T>::index_1 stdout ----
        error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-m64" "-L" "/home/brian/.rustup/toolchains/beta-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "/tmp/rustdoctest.4HYHIIhA9dVC/rust_out.0.o" "-o" "/tmp/rustdoctest.4HYHIIhA9dVC/rust_out" "-Wl,--gc-sections" "-pie" "-nodefaultlibs" "-L" "/mnt2/dev/utah/target/debug/deps" "-L" "/home/brian/.rustup/toolchains/beta-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "-Wl,-Bdynamic" "/mnt2/dev/utah/target/debug/deps/libutah-7ebd1627d8f4b31c.rlib" "/mnt2/dev/utah/target/debug/deps/libndarray_rand-bd14357e2975e921.rlib" "/mnt2/dev/utah/target/debug/deps/libnum-4f5490b7b4313ad0.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_iter-a1bfa3b9618fdfe0.rlib" "/mnt2/dev/utah/target/debug/deps/liberror_chain-51998ce5824b8fe7.rlib" "/mnt2/dev/utah/target/debug/deps/libbacktrace-4bd69dcf892822ef.rlib" "/mnt2/dev/utah/target/debug/deps/libbacktrace_sys-58044020ec178381.rlib" "/mnt2/dev/utah/target/debug/deps/libdbghelp-80abcaa2ef08d9db.rlib" "/mnt2/dev/utah/target/debug/deps/libwinapi-a5898d7aceb63fac.rlib" "/mnt2/dev/utah/target/debug/deps/librustc_demangle-ffe0cd46130fc1c4.rlib" "/mnt2/dev/utah/target/debug/deps/libndarray-a08975650c3bc21a.rlib" "/mnt2/dev/utah/target/debug/deps/libmatrixmultiply-f1139eb0050ad26b.rlib" "/mnt2/dev/utah/target/debug/deps/librawpointer-dc9eeb50dff13477.rlib" "/mnt2/dev/utah/target/debug/deps/libitertools-ce89a6802ef9c3ff.rlib" "/mnt2/dev/utah/target/debug/deps/libeither-f2ccbfaf1069b354.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_complex-2d6fd57f39d31306.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_rational-e1a0d96e4c2f73b6.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_bigint-a0b0a32a0399692f.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_integer-f617fb54b59f25d7.rlib" "/mnt2/dev/utah/target/debug/deps/libnum_traits-51f88ce8f32c0283.rlib" "/mnt2/dev/utah/target/debug/deps/librand-c9d9fbdab2355ee4.rlib" "/mnt2/dev/utah/target/debug/deps/liblibc-13f441799a82421b.rlib" "/mnt2/dev/utah/target/debug/deps/libkernel32-835ed4d4f4dc2d3e.rlib" "/mnt2/dev/utah/target/debug/deps/libcfg_if-5cf5b4dd25466da1.rlib" "/mnt2/dev/utah/target/debug/deps/libcsv-409fb1ac4ede067a.rlib" "/mnt2/dev/utah/target/debug/deps/librustc_serialize-2c0b16b7ceb9fdcb.rlib" "/mnt2/dev/utah/target/debug/deps/libbyteorder-fab4d718c73e8b35.rlib" "-L" "/home/brian/.rustup/toolchains/beta-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-l" "std-dfc89321dbee872e" "/home/brian/.rustup/toolchains/beta-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-a27e0b02588f3694.rlib" "-l" "util" "-l" "dl" "-l" "rt" "-l" "pthread" "-l" "gcc_s" "-l" "pthread" "-l" "c" "-l" "m" "-l" "rt" "-l" "util"
  = note: /mnt2/dev/utah/target/debug/deps/libbacktrace-4bd69dcf892822ef.rlib(backtrace-4bd69dcf892822ef.0.o): In function `backtrace::symbolize::libbacktrace::init_state::{{closure}}':
          /home/brian/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.0/src/symbolize/mod.rs:159: undefined reference to `backtrace_create_state'
          /mnt2/dev/utah/target/debug/deps/libbacktrace-4bd69dcf892822ef.rlib(backtrace-4bd69dcf892822ef.0.o): In function `backtrace::symbolize::libbacktrace::resolve':
          /home/brian/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.0/src/symbolize/mod.rs:159: undefined reference to `backtrace_pcinfo'
          /home/brian/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.0/src/symbolize/mod.rs:159: undefined reference to `backtrace_syminfo'
          collect2: error: ld returned 1 exit status


error: aborting due to previous error

thread 'rustc' panicked at 'Box<Any>', /buildslave/rust-buildbot/slave/beta-dist-rustc-linux/build/src/librustc_errors/lib.rs:453
thread 'rustc' panicked at 'couldn't compile the test', /buildslave/rust-buildbot/slave/beta-dist-rustc-linux/build/src/librustdoc/test.rs:265


failures:
    implement::impl_constructor_generic::DataFrame<T>::columns_0
    implement::impl_constructor_generic::DataFrame<T>::df_iter_0
    implement::impl_constructor_generic::DataFrame<T>::index_0
    implement::impl_constructor_generic::DataFrame<T>::index_1

test result: FAILED. 4 passed; 4 failed; 12 ignored; 0 measured

error: test failed
```

Not on 1.15. __Not on 1.17 nightly__.

cc @pegasos1

**Comment by arielb1** [comment]
@nagisa could you look at this?

**Comment by nagisa** [comment]
> Not on 1.17 nightly.

Fails just fine on nightly for me.

---

The breakage seems to be intended. This “regression” is caused by multiple pieces working together:

1. `utah` crate contains `Cargo.lock`, which locks version of `backtrace-sys` to `0.1.5`;
2. In #39252 libstd stopped exposing its own libbacktrace symbols;
3. `backtrace-sys == 0.1.5` assumes on [linux backtrace symbols are provided by libstd](https://github.com/alexcrichton/backtrace-rs/blob/d0406033841d2df8ed9b63474c86841c86a9f8d8/backtrace-sys/build.rs#L24) (0.1.6 [does not do this anymore](https://github.com/alexcrichton/backtrace-rs/blob/backtrace-sys-0.1.6/backtrace-sys/build.rs#L21)).

Proposed fix: tell the crate author to remove `Cargo.lock` from their packages (and/or run `cargo update`).

**Comment by nagisa** [comment]
cc @alexcrichton perhaps cargo should warn about publishing a package with Cargo.lock?

**Comment by alexcrichton** [comment]
Oh right yes this is a bug in backtrace-sys, fixed in version 0.1.6 (as pointed out by @nagisa)

Heh we actually have some bugs for reading Cargo.lock in published packages, and now we've got bugs for not doing so!

**Comment by pegasos1** [comment]
@brson thanks for opening, @nagisa will remove `Cargo.lock`
